src/metadata.rs

```diff
@@ -147,7 +147,7 @@ impl MetadataBuilder {
 ///
 /// Metadata value can be ascii string or bytes. They are distinguish by the
 /// key suffix, key of bytes value should have suffix '-bin'.
-#[repr(C)]
+#[repr(transparent)]
```

Comment:
seem that all struct with single field using `repr(C)` can be replaced with `repr(transparent)`.